packages/excalidraw/components/App.tsx

```diff
@@ -4265,7 +4263,37 @@ class App extends React.Component<AppProps, AppState> {
         !this.state.selectionElement &&
         !this.state.selectedElementsAreBeingDragged
       ) {
-        const shape = findShapeByKey(event.key);
+        let shape = findShapeByKey(event.key);
```

Comment:
Check if the shape is a valid target shape before iterating over the selected elements.